### PR TITLE
Move SOCA background stager functions to module

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Functions that stage the ocean background are moved from the prep script to a module under ush, in order to facilitate use by the recentering task. Apparently superfluous modules also removed from prep script.

Tested with soca ctests on Hera

This is a step towards addressing https://github.com/NOAA-EMC/GDASApp/issues/912

### CODE REVIEW

File: ush/soca/bkg_utils.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+################################################################################
+#  UNIX Script Documentation Block
+#                      .                                             .
+# Script name:         ush/soca/bkg_utils.py
+# Script description:  Utilities for staging SOCA background
+#
+# Author: Guillaume Vernieres, Andrew Eichmann Org: NCEP/EMC     Date: 2024-02-28
```
Comment (CoryMartin-NOAA) [inline ush/soca/bkg_utils.py]: The future?????

**Review by CoryMartin-NOAA — DISMISSED** [review]

File: ush/soca/bkg_utils.py (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply ush/soca/bkg_utils.py]: I come with a warning

**Review by CoryMartin-NOAA — DISMISSED** [review]

File: scripts/exgdas_global_marine_analysis_prep.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 # Script name:         exufsda_global_marine_analysis_prep.py
 # Script description:  Stages files and generates YAML for UFS Global Marine Analysis
 #
-# Author: Guillaume Vernieres      Org: NCEP/EMC     Date: 2022-03-28
+# Author: Guillaume Vernieres      Org: NCEP/EMC     Date: 2022-03-23
```
Comment (guillaumevernieres) [inline scripts/exgdas_global_marine_analysis_prep.py]: Not why it's edited, but we might as well just remove the header entirely. We'll sort out what to do when moving to the g-w.

**Review by guillaumevernieres — DISMISSED** [review]
Thanks for doing this @AndrewEichmann-NOAA .

**Comment by guillaumevernieres** [comment]
Is it still "work in progress" @AndrewEichmann-NOAA or can I merge your PR?

File: scripts/exgdas_global_marine_analysis_prep.py (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_prep.py]: That's a glitch on my part. I'll take it out.

**Comment by AndrewEichmann-NOAA** [comment]
> Is it still "work in progress" @AndrewEichmann-NOAA or can I merge your PR?

@guillaumevernieres Let me take out the headers and I'll un-draft it

**Comment by CoryMartin-NOAA** [comment]
I deleted the CI directories and will let it try again

**Comment by AndrewEichmann-NOAA** [comment]
The log `/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/935/global-workflow/sorc/logs/build_gdas.log` has the following:

```
-- [download 51% complete]
CMake Error at gdas/test/CMakeLists.txt:8 (file):
  file DOWNLOAD HASH mismatch

    for file: [/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/935/global-workflow/sorc/gdas.cd/build/gdas/test/gdasapp-fix-0412bf.tgz]
      expected hash: [0412bf1037a28bb66e849ee9e03aadab4d01cd73b6cbc25dfb6a293e694006dc]
        actual hash: [f070c383997bb90b02c36fbcbd50f995b313270684c327ebbf00b73b2b4b86ba]
             status: [28;"Timeout was reached"]



cmake -E tar: error: truncated gzip input
ctest for JGLOBAL_PREP_OCEAN_OBS
```
I don't do `git add -a`, did a bad has slip through anyway?

**Comment by guillaumevernieres** [comment]
Your pr isn't the issue @AndrewEichmann-NOAA . The tests were fine (except for the known failures) before you triggered the ci again. I tend to not re-tests if all I change is a comment.

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Comment by RussTreadon-NOAA** [comment]
Any more GDASApp PRs we want to merge into `develop` before I update the g-w `gdas.cd` hash?

**Comment by guillaumevernieres** [comment]
> Any more GDASApp PRs we want to merge into `develop` before I update the g-w `gdas.cd` hash?

Thanks for checking @RussTreadon-NOAA . We're good, even with the commit before this PR.

**Pull request merged**